src/client.ts

```diff
@@ -463,6 +475,225 @@ export class McpdClient {
+    const healthyServers = await this.#getHealthyServers(servers);
+
+    // Fetch resource templates from all healthy servers in parallel.
+    const results = await Promise.allSettled(
+      healthyServers.map(async (serverName) => ({
+        serverName,
+        templates: await this.#getResourceTemplatesByServer(serverName),
+      })),
+    );
+
+    // Process results and transform template names.
+    const allTemplates: ResourceTemplate[] = results.flatMap((result) => {
+      if (result.status === "fulfilled") {
+        const { serverName, templates } = result.value;
+        return templates.map((template) => ({
+          ...template,
+          name: `${serverName}__${template.name}`,
+          _serverName: serverName,
+          _templateName: template.name,
+        }));
+      } else {
+        console.warn(
+          `Failed to get resource templates for server:`,
+          result.reason,
+        );
+        return [];
+      }
+    });
+
+    return allTemplates;
+  }
+
+  /**
+   * Read resource content using a namespaced resource name.
+   *
+   * This method uses an internal cache to map namespaced names to their server and URI.
+   * If the resource is not found in cache, it automatically calls getResources() to
+   * populate the cache and retries the lookup.
+   *
+   * @param namespacedName - The namespaced resource name (serverName__resourceName)
+   * @returns Array of resource contents (text or blob)
+   * @throws {Error} If the resource is not found after populating the cache
+   * @throws {ServerNotFoundError} If the specified server doesn't exist
+   * @throws {ServerUnhealthyError} If the server is not healthy
+   * @throws {ConnectionError} If unable to connect to the mcpd daemon
+   * @throws {TimeoutError} If the request times out
+   * @throws {McpdError} If the request fails
+   *
+   * @example
+   * ```typescript
+   * // Read content by namespaced name - cache populated automatically if needed
+   * const contents = await client.readResource("github__readme");
+   * for (const content of contents) {
+   *   if (content.text) {
+   *     console.log(content.text);
+   *   } else if (content.blob) {
+   *     console.log('Binary content (base64):', content.blob.substring(0, 50) + '...');
+   *   }
+   * }
+   *
+   * // You can also pre-populate the cache for efficiency if reading multiple resources
+   * await client.getResources();
+   * const readme = await client.readResource("github__readme");
+   * const changelog = await client.readResource("github__changelog");
+   * ```
+   */
+  async readResource(namespacedName: string): Promise<ResourceContent[]> {
+    // Look up resource in cache.
+    let cached = this.#resourceCache.get(namespacedName);
+
+    // If not in cache, populate it by calling getResources().
+    if (!cached) {
+      await this.getResources();
+      cached = this.#resourceCache.get(namespacedName);
```

Comment:
feels weird to read `getResources` without using its returned value, can't we just do this?


```suggestion
      const resources = await this.getResources();
```